### DISCUSSION TITLE
Send keyboard keys in remote session

### DISCUSSION THREAD

**Discussion by adispy** [Ylianst/MeshCentral#5548] [Ideas]
Hi everyone,
Hope this is the correct forum, but just as a feature it will be nice to be able to send the keyboard to the remote endpoint once in a session. Like for example if I am doing Alt+Tab to be able to change screens in the remote session not on my machine where the browser is opened.

Thanks,

**Comment by dinger1986** [comment]
Down the bottom left corner of your browser when you are connected there's keys which can be selected from the drop down and then click send, you can also customise that list using left click. And edit.

I thought this was in the docs but couldn't find it either.

**Comment by adispy** [comment]
I was not talking about that, not sending keys by pressing the buttons from the left corner.

When you are in a remote session and want to use the Run command on Windows which is Win+R to open in the remote session, not on the host.
The remote session should capture the keyboard entirely like AnyDesk and Team Viewer does.

**Reply by dinger1986** [reply]
I understand but with mesh being web based this doesnt work, you can use this instead

![image](https://github.com/Ylianst/MeshCentral/assets/7244447/a3f25fba-cd3b-4c95-b9cb-10c3410695bd)

**Comment by adispy** [comment]
I understand. Strange tough, because I use VMware ESXi products which are also web based, and when opening a console for a virtual machine the keyboard is captured there.
Thought it's the same, I'm not a web programmer.

Thanks anyway,

**Reply by dinger1986** [reply]
I think vmware captures the keyboard differently

**Comment by si458** [comment]
CHATGPT even agrees, using ALT+TAB isnt possible in web browsers so not sure how VMWARE esxi web ui is doing it?

Me: how do i capture the ALT+TAB event in HTML/JS and NOT call the windows events

CHATGPT: 
Capturing the ALT+TAB event in a web page using HTML/JS is not straightforward because the browser intentionally restricts access to certain system-level events for security reasons. The ALT+TAB combination is one of those events that browsers typically don't expose to JavaScript.

**Reply by silversword411** [reply]
Esci has a viewer it can launch.

Probably using that

**Comment by panreyes** [comment]
It looks like this feature has already been solved in the current version, as pressing Shift while clicking on the Full screen button will enable what the OP is requesting :)
